### ISSUE TITLE
`buildBeforeRestart` does not trigger rebuild of project in tests

### ISSUE THREAD

**Issue by jyliuu** [rust-lang/rust-analyzer#18763]
**rust-analyzer version**: `rust-analyzer version: 0.3.2228-standalone (fa4a40bbe8 2024-12-22)`

**rustc version**: `rustc 1.83.0 (90b35a623 2024-11-26)`

**editor or extension**: VSCode (Please provide the version of the rust-analyzer extension you are using)

**relevant settings**:
```json
{
    "rust-analyzer.debug.buildBeforeRestart": true
}
```

**code snippet to reproduce**:
```rust
// Run Test | Debug
#[test]
fn my_test() {
   ...
}
```

**Description of the Issue:**

When debugging a Rust test in VSCode by clicking the "Debug" button above the test function, and then subsequently hitting the "Restart" button in the debug controls, rust-analyzer does not appear to recompile the code. This means that if changes have been made to the code since the initial debug launch, those changes are not reflected when the debug session is restarted.

The `rust-analyzer.debug.buildBeforeRestart` setting is set to `true`, which according to the documentation, should trigger a rebuild before restarting a debug session. However, this is not the observed behavior.

**Steps to Reproduce:**

1. Have a Rust project with a test function like the example provided.
2. Ensure `"rust-analyzer.debug.buildBeforeRestart": true` is set in your VSCode settings (preferably workspace settings).
3. Click the "Debug" button above the test function to start a debug session.
4. Make a change to the code that would affect the test's execution (e.g., modify the `TreeGridParams`, change the logic in `setup_data`, etc.).
5. Click the "Restart" button in the debug controls.
6. Observe that the debug session starts without recompiling the code, and the changes made in step 4 are not reflected in the restarted debug session.

**Expected Behavior:**

When the "Restart" button is clicked during a debug session, and `rust-analyzer.debug.buildBeforeRestart` is set to `true`, rust-analyzer should trigger a recompilation of the project before starting the new debug session, ensuring that the latest code changes are included.

**Comment by jyliuu** [comment]
Seems like this is related to https://github.com/microsoft/vscode/issues/193039, it might be a LLDB issue.
